tc/core/utils/time.h

```diff
@@ -18,7 +18,61 @@
 #include <chrono>
 
 namespace tc {
-using Duration = std::chrono::high_resolution_clock::duration;
+struct Duration {
+ private:
+  explicit Duration(size_t us) : val_(us) {}
+  explicit Duration(std::chrono::microseconds us) : val_(us) {}
+
+ public:
+  Duration() : val_(std::chrono::microseconds::max()) {}
+  Duration(const Duration& d) : val_(d.val_) {}
+
+  static inline Duration fromStartStop(
+      std::chrono::time_point<std::chrono::system_clock> begin,
+      std::chrono::time_point<std::chrono::system_clock> end) {
+    return Duration(
+        std::chrono::duration_cast<std::chrono::microseconds>(end - begin));
+  }
+  static inline Duration fromMicroSeconds(size_t us) {
+    return Duration(us);
+  }
+  static inline Duration zero() {
+    return Duration(0);
+  }
+  static inline Duration max() {
+    return Duration(std::chrono::microseconds::max());
+  }
+  inline size_t count() {
```

Comment:
`count` doesn't mean much on something called `Duration`.
Given that you also have a `fromMicroSeconds`, maybe `inMicroSeconds`?